### PR TITLE
add `head` decorator

### PR DESCRIPTION
**Pull request by Goldziher**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

This PR adds a `head` decorator.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -142,7 +143,6 @@ max-line-length = "120"
 ignored-argument-names = "args|kwargs|_|__"
 
 [tool.pylint.BASIC]
-good-names = "_,__,i,e,k,v,fn,fs,get,post,put,patch,delete,route,asgi,websocket,Dependency,Body,Parameter,HandlerType,ScopeType,Auth,User,it"
```
Comment (Goldziher) [inline pyproject.toml:145]: removed because this is becoming very long, and frankly this inspection is a nuisance.

Comment (peterschutt) [reply pyproject.toml:145]: sgtm.

File: starlite/handlers/http.py [hunk 2]
```diff
@@ -211,6 +212,42 @@ async def handler(data: Any, plugins: List["PluginProtocol"], **kwargs: Any) ->
...
+    Returns:
+        A status code integer.
+    """
+    if HttpMethod.POST in http_methods:
+        return HTTP_201_CREATED
+    if HttpMethod.DELETE in http_methods:
+        return HTTP_204_NO_CONTENT
+    return HTTP_200_OK
+
+
```
Comment (peterschutt) [inline starlite/handlers/http.py:250]: Maybe a `handlers.utils` module for these?

File: starlite/handlers/http.py [hunk 3]
```diff
@@ -757,6 +791,158 @@ def __init__(
...
+            **kwargs,
+        )
+
+    def _validate_handler_function(self) -> None:
+        """Validates the route handler function once it is set by inspecting
+        its return annotations."""
+        super()._validate_handler_function()
+
+        # we allow here File and FileResponse because these have special setting for head responses
+        if self.signature.return_annotation not in {None, "None", File, "File", FileResponse, "FileResponse"}:
```
Comment (peterschutt) [inline starlite/handlers/http.py:940]: What if it is a subclass of `File` or `FileResponse`?

**Review by peterschutt — APPROVED** [review]

File: starlite/handlers/http.py (hunk 2 above)
Comment (Goldziher) [reply starlite/handlers/http.py:250]: I dont think we will ever use them outside this module.

Comment (peterschutt) [reply starlite/handlers/http.py:250]: and in the tests.. not a blocker, just a suggestion.

File: starlite/handlers/http.py (hunk 3 above)
Comment (Goldziher) [reply starlite/handlers/http.py]: fixed

Comment (peterschutt) [reply starlite/handlers/http.py]: Unless it's a forward ref of a subclass :wink:

**Pull request merged**